### PR TITLE
test: created PyTest harness for SHACL  testing

### PR DESCRIPTION
**Pull request by Markus92**
Initial pytest harness for the SHACLs

Right now I used parametrized tests for the core shapes. It'll be possible to add more tests, also tests that have to fail, xpass/xfail tests etc.

### CODE REVIEW

File: tests/test_shacl.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@pytest.mark.parametrize(
+    ("shacl", "example"),
+    [
+        ("Catalog", "example-catalog"),
+        ("Dataset", "example-dataset"),
+        ("Distribution", "example-distribution"),
+    ],
+)
+def test_core_shapes(shacl, example):
+    shacl_graph = rf"Formalisation(shacl)/Core/PiecesShape/{shacl}.ttl"
```
Comment (PatrickDekkerHealthRI) [inline tests/test_shacl.py:14]: Can you move the paths out of this function ? I prefer to have additional functions getTest() and getShacl().

File: tests/test_shacl.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+@pytest.mark.parametrize(
+    ("shacl", "example"),
+    [
+        ("Catalog", "example-catalog"),
+        ("Dataset", "example-dataset"),
+        ("Distribution", "example-distribution"),
+    ],
+)
+def test_core_shapes(shacl, example):
```
Comment (PatrickDekkerHealthRI) [inline tests/test_shacl.py:13]: The name of the functions is not what you do.... i think the name shoudl be test_graph_conforms_shacl or something like that..

**Review by PatrickDekkerHealthRI — CHANGES_REQUESTED** [review]
Would be nice if there is bit more explanation on 

1. how to run the test locally
2.  How to add more test case. 
3. One example of failing validation would be nice:


I do have some code that check that for every shacl files at least an test file exists.. It matched the shacl on name so i was not really happy with that if can come up with a better way that would be nice...

File: tests/test_shacl.py (hunk 2 above)
Comment (Markus92) [reply tests/test_shacl.py:13]: This is the name that shows up in the rest report. I think `test_graph_conforms_shacl` might not be super descriptive, but open to suggestions.

File: tests/test_shacl.py
Comment (kburger) [inline tests/test_shacl.py:1]: I like that it's small and concise, and relatively easy to add new tests for newcomers. Patrick's comments are valid, but I'd be OK with refactoring later(ish).

**Review by kburger — DISMISSED** [review]

File: tests/test_shacl.py (hunk 1 above)
Comment (Markus92) [reply tests/test_shacl.py]: Done

**Comment by Markus92** [comment]
In the new version, I renamed the action to be more clear than just "ci" and made sure it runs on every PR (not just push, as that might skip it for PR's coming from different forks). I also added some basic documentation on how to run the test cases.

**Review by PatrickDekkerHealthRI — DISMISSED** [review]
Good to go!

**Review by PatrickDekkerHealthRI — APPROVED** [review]
Ok

**Pull request merged**